src/unix/process.c

```diff
@@ -275,22 +275,22 @@ static void uv__process_child_init(
     use_fd = pipes[fd][1];
     if (use_fd < 0 || use_fd >= fd)
       continue;
-    pipes[fd][1] = fcntl(use_fd, F_DUPFD, stdio_count);
+    pipes[fd][1] = fcntl(use_fd, F_DUPFD_CLOEXEC, stdio_count);
     if (pipes[fd][1] == -1)
       uv__write_errno(error_fd);
   }
 
   for (fd = 0; fd < stdio_count; fd++) {
-    close_fd = pipes[fd][0];
+    close_fd = -1;
     use_fd = pipes[fd][1];
 
     if (use_fd < 0) {
       if (fd >= 3)
         continue;
       else {
-        /* redirect stdin, stdout and stderr to /dev/null even if UV_IGNORE is
-         * set
-         */
+        /* Redirect stdin, stdout and stderr to /dev/null even if UV_IGNORE is
+         * set. */
+        uv__close_nocheckstdio(fd); /* Free up fd, if it happens to be open. */
         use_fd = open("/dev/null", fd == 0 ? O_RDONLY : O_RDWR);
```

Comment:
Do you think `O_CLOEXEC` could also be used here?

Reply:
yeah, I agree it could, but didn't seem beneficial, and there's a slight chance some kernel won't support it